### ISSUE TITLE
Making imaginary modules real with webpack and babel

### ISSUE THREAD

**Issue by markbrocato** [charmCityJs/talks#99]
I recently wrote [extjs-reactor](https://github.com/sencha/extjs-reactor), which turns Ext JS into a component library for React.  Why is this interesting?

1. Ext JS does not export ES6 modules (or even commonJS modules), so I had to write a babel transform to make it look like it does.
2. Ext JS is a huge library which needs to be tree-shaken for production use.  Due to #1, nothing other than our own in-house build tool (Sencha Cmd) can tree-shake it.  So I had to write a webpack plugin to integrate Sencha Cmd into the typical react workflow.
3. In order to integrate Ext JS and react with minimal code, I had to dig into the react rendering API.  Cool stuff!

Basically the point of this talk is to give the audience some insight into some crazy real-world magic hacks that you can pull off if you take the time to learn babel, webpack, and react internals.  Might inspire some outside the box thinking - but probably will lead to a whole lot of awful hacks being written in Baltimore :)

P.S.  The focus of the talk will be webpack, babel, and react.  Ext JS will only be mentioned to introduce the problem I had to solve.

**Comment by jasonrhodes** [comment]
Love this idea, Mark. :)

**Comment by esauter5** [comment]
@markbrocato are you available to give this on January 4?

**Comment by markbrocato** [comment]
Yes!

**Comment by esauter5** [comment]
Great :)

How long do you expect this to be?

**Comment by markbrocato** [comment]
Ideally about 15 minutes.

**Comment by esauter5** [comment]
Perfect. You're all set. See you then!

**Comment by esauter5** [comment]
Thanks for speaking, @markbrocato!
